Code cleanup (replaces #8, #34, #39, #40, fixes #35, #38)

ConorIA opened the pull request:
This pull request implements the changes suggested in #34, as well as some additional linting from python-flake8. 

This also includes the changes in #39 and #40, so merging this could close out all three of the other PRs. 

I probably should have waited for final feedback on #40, but I am happy to make changes to this PR based on the outcome of #40.

tommeagher commented:
Hi @ConorIA. This all looks awesome. Thanks also for incorporating all the fixes @00dani had made. I'm inclined to merge this. The only thing missing I think is a little explanation of the new local_settings.
Would you be able to add a little bit to the "Configuring" section in the README? 
Otherwise, I think this is good to go.

ConorIA commented:
Alright, done! I did make two small non-README changes. 

1. I set the `scrape` function to check the last URL, so that we only download a page once if a user wants to search it multiple times, as alluded to in the README this gives us less opportunities to hit an HTTP error or get flagged for hitting the same page multiple times. 
2. I added the space suggested in #8, which I had implemented in my production script, but had forgotten to copy back to my forked repo. Talk about an omnibus PR! 😜

tommeagher commented on ebooks.py:
```diff
@@ -34,119 +38,143 @@ def entity(text):
-    tweet.text = re.sub(r'\s+\(?(via|says)\s@\w+\)?', '', tweet.text) # remove attribution
+    tweet.text = re.sub(r'\b(RT|MT) .+', '', tweet.text)  # take out anything after RT or MT
+    tweet.text = re.sub(r'(\#|@|(h\/t)|(http))\S+', '', tweet.text)  # Take out URLs, hashtags, hts, etc.
+    tweet.text = tweet.text.replace('\n', '')  # take out new lines.
```
This should be 
```tweet.text = re.sub('\s+', ' ', tweet.text)```
per #35

ConorIA replied:
Done.

tommeagher commented on markov.py:
```diff
@@ -68,7 +69,7 @@ def generate_sentence(self):
             for word in new_res:
                 sentence += word + " "
-            sentence += res[-2] + res[-1]
+            sentence += res[-2] + " " + res[-1]
```
This adds an unnecessary space before the punctuation of every generated tweet. 
Like this: "Why can't I fix this via ." and "ALL THE THINGS AND WE WILL ."
This should be reverted to exclude this extra space.

ConorIA replied:
Fixed. Please test. I'm not sure how Pythonic the fix is, but it seems to work!

tommeagher replied:
Not very Pythonic, but it seems to work fine, so it's good by me. Thank you!

tommeagher commented on ebooks.py:
```diff
@@ -34,119 +38,143 @@ def entity(text):
+            soup = BeautifulSoup(page, 'html.parser')
+        hits = soup.find_all(web_context[i], attrs=web_attributes[i])
+        for hit in hits:
+            tweet = str(hit.text).strip()
```
This returns a UnicodeEncodeError in Python2. So there should be an if statement here to catch the exception.

ConorIA replied:
Handled! I think the `UnicodeDecodeError` might be superfluous. Let me know if you'd like me to remove.

tommeagher commented on local_settings_example.py:
```diff
@@ -2,17 +2,21 @@
-STATIC_TEST = False #Set this to True if you want to test Markov generation from a static file instead of the API.
-TEST_SOURCE = ".txt" #The name of a text file of a string-ified list for testing. To avoid unnecessarily hitting Twitter API. You can use the included testcorpus.txt, if needed.
-TWEET_ACCOUNT = "" #The name of the account you're tweeting to.
+SOURCE_ACCOUNTS = [""]  # A list of comma-separated, quote-enclosed Twitter handles of account that you'll generate tweets based on. It should look like ["account1", "account2"]. If you want just one account, no comma needed.
```
We should mention that you *must* include at least one twitter account here, or the script fails with an ```IndexError``` on line 139 of ebooks.py

ConorIA replied:
Handled in code by first testing to make sure that `SOURCE_ACCOUNTS` is not empty.

ConorIA commented on ebooks.py:
```diff
@@ -34,119 +38,143 @@ def entity(text):
-    tweet.text = re.sub(r'\xe9', 'e', tweet.text) #take out accented e
+    for item in htmlsents:
+        tweet.text = tweet.text.replace(item, entity(item))
+    tweet.text = tweet.text.replace('\xe9', 'e')  # take out accented e
```
Insofar as the change was to make the code "cleaner", I think a try-except block is considerable "messier". Reverted.

tommeagher commented:
This all looks swell. Thanks, @ConorIA for everything. Merging now.
